src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java

```diff
@@ -11,31 +12,31 @@
 	 * This should always have the same return value.
 	 * @return Channel Identifier
 	 */
-	public abstract String getChannelIdentifier();
+	public abstract Id getChannelIdentifier();
 	
 	/**
-	 * Called when this channel recieves a packet
+	 * Called when this channel recieves a packet.
 	 * @param arg The local PacketHandler
 	 * @param data The data of the packet
 	 */
 	public abstract void onRecieve(PacketHandler arg, byte[] data);
 	
 	/**
-	 * Server packet send method
+	 * Server packet send method.
 	 * @param data The data to send
 	 * @param player The player to send the data to
 	 */
 	public void send(byte[] data, ServerPlayer player) {
-		player.field_255.method_835(new PluginMessagePacket(getChannelIdentifier(), data));
+		player.field_255.method_835(new PluginMessagePacket(getChannelIdentifier().toString(), data));
 	}
 
 	/**
-	 * Client packet send method
+	 * Client packet send method.
 	 * @param data The data to send
 	 * @param mc The local Minecraft
 	 */
 	public void send(byte[] data, Minecraft mc) {
-		mc.method_2145().sendPacket(new PluginMessagePacket(getChannelIdentifier(), data));
+		mc.method_2145().sendPacket(new PluginMessagePacket(getChannelIdentifier().toString(), data));
```

Comment:
change the ctor

Reply:
?

Reply:
change the constructor of plugin message packet instead of tostringing it

Reply:
no

Reply:
do it or I'll slap you

Reply:
false

Reply:
👏 do it or I'll slap u again

Reply:
http://i.imgur.com/h2aHRVa.png